networkconfig.py

```diff
@@ -3,201 +3,304 @@
+                    Be careful about security!
+                </span>
+            </h5>
+            <hr />
+            <h2 style="color: #2e6c80;">
+                Some useful infos:
+            </h2>
+            <ul>
+                <li>
+                    Original code from <a href="https://github.com/cpopp/MicroPythonSamples"
+                        target="_blank" rel="noopener">cpopp/MicroPythonSamples</a>.
+                </li>
+                <li>
+                    This code available at <a href="https://github.com/tayfunulu/WiFiManager"
+                        target="_blank" rel="noopener">tayfunulu/WiFiManager</a>.
+                </li>
+            </ul>
+        </html>
+    """ % dict(filename=NETWORK_PROFILES))
+    send_response(client, "\n".join(response))
+
 
 def handle_configure(client, request):
-	match = ure.search("ssid=([^&]*)&password=(.*)", request)
-
-	if match is None:
-		send_response(client, "Parameters not found", status_code=400)
-		return (False)
-	# version 1.9 compatibility
-	try:
-		ssid = match.group(1).decode("utf-8").replace("%3F","?").replace("%21","!")
-		password = match.group(2).decode("utf-8").replace("%3F","?").replace("%21","!")
-	except:
-		ssid = match.group(1).replace("%3F","?").replace("%21","!")
-		password = match.group(2).replace("%3F","?").replace("%21","!")
-	
-	
-	if len(ssid) == 0:
-		send_response(client, "SSID must be provided", status_code=400)
-		return (False)
-
-	if do_connect(ssid, password):
-		response_footer = """
-		<html>
-		<center><br><br>
-		<h1 style="color: #5e9ca0; text-align: center;"><span style="color: #ff0000;">:) YES, Wi-Fi Configured to """+ssid+"""</span></h1>
-		<br><br>"""
-		send_response(client, response_footer)
-		try:
-			fo = open("passwd.dat","r")
-			ex_data = fo.read()
-			fo.close()
-			fo = open("passwd.dat","w")
-			ex_data = ssid+";"+password+"\n"+ex_data
-			fo.write(ex_data)
-			fo.close()
-		except:
-			fo = open("passwd.dat","w")
-			fo.write(ssid+";"+password+"\n")
-			fo.close()
-		
-		return (True)
-	else:
-		response_footer = """
-		<html>
-		<center>
-		<h1 style="color: #5e9ca0; text-align: center;"><span style="color: #ff0000;">Wi-Fi Not Configured to """+ssid+"""</span></h1>
-		<br><br>
-		<form>
-			<input type="button" value="Go back!" onclick="history.back()"></input>
-		</form></center></html>
-		"""
-		send_response(client, response_footer )
-		return (False)
+    match = ure.search("ssid=([^&]*)&password=(.*)", request)
+
+    if match is None:
+        send_response(client, "Parameters not found", status_code=400)
+        return False
+    # version 1.9 compatibility
+    try:
+        ssid = match.group(1).decode("utf-8").replace("%3F", "?").replace("%21", "!")
+        password = match.group(2).decode("utf-8").replace("%3F", "?").replace("%21", "!")
+    except:
+        ssid = match.group(1).replace("%3F", "?").replace("%21", "!")
+        password = match.group(2).replace("%3F", "?").replace("%21", "!")
```

Comment:
these try/except blocks are still rather ugly.
not sure what you're catching here, is it just bytes vs. str or something else also?